### PR TITLE
feat: retrieve pull-secret using Red Hat SSO account

### LINKED ISSUE #207: Implement pull secret retrieval through API and integrate into workflow

### PR DESCRIPTION
**Pull request by dgolovin**
Fix #207.

This PR adds:
1. Dependency to Red Hat Account extension
2. Confirmation to allow login is requested
3. If allowed workflow goes though login in browser and retrieval of pull request using access token
4. If cancelled old workflow goes as fallback

Considered other implementation options:
1. Show Quick Picks instead of info message
2. Always do it with Red Hat Account and use manual input as backup. The problem is when login sequence is started there is not way to cancel it.

Depends on:

- [x] https://github.com/containers/podman-desktop/pull/7443
- [ ] https://github.com/containers/podman-desktop/pull/7466

### CODE REVIEW

File: src/crc-start.ts [hunk 1]
```diff
@@ -91,12 +92,42 @@ export async function startCrc(
...
-    ignoreFocusOut: true,
-  });
+  let pullSecret: string;
+  const howToPull = await extensionApi.window.showInformationMessage(
+    'To pull container images from the registry, a *pull secret* is necessary. You can login into your Red Hat SSO account to get it configured automatically or manually copy it from browser and paste when requested.',
+    'Sign in with Red Hat SSO',
+    'Configure manually',
+  );
+  if (howToPull) {
+    if (howToPull === 'Use Red Hat SSO Account') {
```
Comment (jeffmaury) [inline src/crc-start.ts:102]: This does not seem to match the label as I am always redirected to the manual pull secret window

**Review by jeffmaury — CHANGES_REQUESTED** [review]

File: src/crc-start.ts (hunk 1 above)
Comment (dgolovin) [reply src/crc-start.ts]: Updated to match

**Review by jeffmaury — COMMENTED** [review]
Works for me but I'm wondering we should enfore SSO use as manual retrieval is too complex. @slemeur WDYT ?

**Comment by slemeur** [comment]
Yes, we should enforce the use of SSO.
We could keep an option if the user wants to bring the pull secret - but that should not be the default path.

**Comment by dgolovin** [comment]
This issue https://github.com/containers/podman-desktop/issues/6632 should also be fixed to make workflow better. Right now the browser just pops up and one has no idea why.

**Comment by dgolovin** [comment]
> We could keep an option if the user wants to bring the pull secret - but that should not be the default path.

The option is already exists - pointing to file with pull-secret in extension settings.

image

**Review by jeffmaury — CHANGES_REQUESTED** [review]
Fix packaging errors

**Comment by dgolovin** [comment]
depends on https://github.com/containers/podman-desktop/pull/7443.

**Comment by dgolovin** [comment]
Depends on
- [ ] #231

**Pull request merged**